### ISSUE TITLE
Docker container stopped unexpectedly when updating

### ISSUE THREAD

**Issue by rfgamaral** [containrrr/watchtower#237]
I have watchtower running on my Synology NAS and most of the times it finds new images and tries to update them, I get a system notification saying that the updated container stopped unexpectedly:

![image](https://user-images.githubusercontent.com/96476/47428269-a6481280-d78a-11e8-8235-4118f353b9e2.png)

When I stop the containers manually (either from Synology GUI or CLI) I don't get this system message, the container stops properly.

Is there anything that can be done to improve this behavior?

**Comment by SuperSandro2000** [comment]
Some logs would be appreciated

**Comment by rfgamaral** [comment]
Which logs exactly do you need?

**Comment by SuperSandro2000** [comment]
I think I missinterpretated your issue. 
Does the container really stop and does not restart? If yes some logs of that container would be good to know why it stopped.
And then watchtower logs especially the section right before the message appeared. If everything just runs fine then you probably need to add a grace time until you get that notification cause the container needs to restart to update the image.

**Comment by rfgamaral** [comment]
The container stops and then gets recreated, watchtower is working as intended the only thing is that it doesn't seem to stop containers gracefully:

![image](https://user-images.githubusercontent.com/96476/47565787-83039b80-d921-11e8-9df8-8830e9be0370.png)

**Comment by SuperSandro2000** [comment]
According to [GNU](https://www.gnu.org/software/libc/manual/html_node/Termination-Signals.html) this is gracefully.
```
Macro: int SIGTERM
The SIGTERM signal is a generic signal used to cause program termination. Unlike SIGKILL, this signal can be blocked, handled, and ignored. It is the normal way to politely ask a program to terminate.

The shell command kill generates SIGTERM by default.
```

**Comment by rfgamaral** [comment]
Well, I'm not saying `SIGTERM` is not gracefully but when watchtower stops my containers, my Synology sends a notification saying the container stopped unexpectedly but when I manually stop the containers, I don't get that notification.

**Comment by SuperSandro2000** [comment]
Sounds more like an issue with your Synology rather than watch tower.

**Comment by rfgamaral** [comment]
How can it be an issue with my Synology when this **only** happens when watchtower stops the containers?

For testing, I just stopped one of my containers with `docker container stop bedbd90199ba` and my Synology didn't give me any notification, the container stopped normally.

Why not stop the containers with `docker container stop` instead of `SIGTERM`?

**Comment by SuperSandro2000** [comment]
Docker stop uses sigterm, too. I could imagine that watchtower sends both but I don't really know.

**Comment by darknode** [comment]
It looks like watchtower "stops" containers with `client.api.ContainerKill(bg, c.ID(), signal)`. This might be a source of Synology notifications.

https://github.com/v2tec/watchtower/blob/8197bb669d2f3c992a094572c455cc71f6b866a8/container/client.go#L88-L95

**Comment by simonaronsson** [comment]
As far is I can tell, this is not a watchtower issue but rather something with how synology checks whether a stop was graceful or not. You could try experimenting with different kill signals, but other than that; id suggest you let synology know.

**Comment by anthonyvallone** [comment]
@simskij I have looked at the docker api and in this instance we should be using `func (cli *Client) ContainerStop(ctx context.Context, containerID string, options container.StopOptions) error` rather than `func (cli *Client) ContainerKill(ctx context.Context, containerID, signal string) error`

In the instance that the graceful stop doesnt work within the specified timeout period it will then perform a SIGKILL.

I have modified the code to do this. I will test and submit a pull request when done.

```
if c.IsRunning() {
	log.Infof("Stopping %s (%s) with %s", c.Name(), shortID, signal)
	if err := client.api.ContainerStop(bg, idStr, c.StopOptions{ Signal: "SIGKILL", Timeout: &timeout,}); err != nil {
		return err
	}
}
```

@rfgamaral hopefully this fixes the issue as im experiencing it too :)
